client/src/components/Legend/Legend.tsx

```diff
@@ -1,4 +1,4 @@
-import React from "react";
+import React, { useEffect, useRef } from "react";
```

Comment:
Legend now uses `useEffect`/refs to rebuild the continuous gradient when autosave changes colorable annotations. If you see flicker, this is the place to tweak.